dot-object/dot-object.d.ts

```diff
@@ -0,0 +1,162 @@
+// Type definitions for Dot-Object v1.4.1
+// Project: https://github.com/rhalff/dot-object
+// Definitions by: Niko Kovačič <https://github.com/nkovacic>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
+declare namespace DotObject {
+    interface Dot {
+        new(separator: string): Dot;
```

Comment:
This isn't correct unless you can chain an arbitrary number of `new`s together, which I doubt. The construct signature should be in its own type so this isn't recursive.